unixsocket/src/main/scala/fs2/io/unixsocket/UnixSocket.scala

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright (c) 2013 Functional Streams for Scala
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package fs2.io.unixsocket
+
+import cats.implicits._
+import cats.effect._
+import fs2._
+import fs2.io.net.Socket
+import jnr.unixsocket._ // This is where the fun begins
+import jnr.unixsocket.impl.AbstractNativeSocketChannel
+import java.nio.{Buffer, ByteBuffer}
+import java.nio.file._
+import cats.effect.std._
+import java.net.InetSocketAddress
+
+object UnixSocket {
+
+  def client[F[_]: Async](address: UnixSocketAddress): Resource[F, Socket[F]] =
+    Resource
+      .eval(Sync[F].delay(UnixSocketChannel.open(address)))
+      .flatMap(makeSocket[F](_))
+
+  def server[F[_]](
+      address: UnixSocketAddress
+  )(implicit
+      F: Async[F]
+  ): Resource[F, Stream[F, Resource[F, Socket[F]]]] = {
+    def setup = fs2.io.file
+      .Files[F]
+      .exists(Paths.get(address.path))
+      .ifM(
+        F.raiseError(
+          new RuntimeException(
+            "Socket Location Already Exists, Server cannot create Socket Location when location already exists."
+          )
+        ),
+        F.blocking {
+          val serverChannel = UnixServerSocketChannel.open()
+          serverChannel.configureBlocking(false)
+          val sock = serverChannel.socket()
+          sock.bind(address)
+          serverChannel
+        }
+      )
+
+    def cleanup(sch: UnixServerSocketChannel): F[Unit] =
+      F.blocking {
+        if (sch.isOpen) sch.close()
+        if (sch.isRegistered()) println("Server Still Registered")
```

Comment:
Not sure how you want to manage if this occurs. It shouldn't, but it has.

Also could delete when we are done if this is not the case.